engine/resources/mount.go

```diff
@@ -0,0 +1,715 @@
+		return errwrap.Wrapf(err, "error reloading systemd daemon")
+	}
+
+	// systemctl restart local-fs.target
+	if _, err := conn.RestartUnit("local-fs.target", "fail", connChan); err != nil {
+		return errwrap.Wrapf(err, "error restarting local-fs.target")
+	}
+	select {
+	case msg, ok := <-connChan:
+		if !ok {
+			return fmt.Errorf("channel closed unexpectedly")
+		}
+		if msg != "done" {
+			return fmt.Errorf("unexpected systemd return string: %s", msg)
+		}
+	}
+
+	// systemctl restart remote-fs.target
+	if _, err := conn.RestartUnit("remote-fs.target", "fail", connChan); err != nil {
+		return errwrap.Wrapf(err, "error restarting remote-fs.target")
+	}
+	select {
+	case msg, ok := <-connChan:
+		if !ok {
+			return fmt.Errorf("channel closed unexpectedly")
+		}
+		if msg != "done" {
+			return fmt.Errorf("unexpected systemd return string: %s", msg)
+		}
+	}
+	return nil
+}
+
+// evalSpec resolves the device from the supplied spec, i.e. it follows the
+// symlink, if any, from the provided uuid, label, or path.
+func evalSpec(spec string) (string, error) {
+	var path string
+	m := &fstab.Mount{}
+	m.Spec = spec
+
+	switch m.SpecType() {
+	case fstab.UUID:
+		path = diskByUUID + m.SpecValue()
+	case fstab.Label:
+		path = diskByLabel + m.SpecValue()
+	case fstab.PartUUID:
+		path = diskByPartUUID + m.SpecValue()
+	case fstab.PartLabel:
+		path = diskByPartLabel + m.SpecValue()
+	case fstab.Path:
+		path = m.SpecValue()
+	default:
+		return "", fmt.Errorf("unexpected spec type: %s", m.SpecType())
+	}
+
+	return filepath.EvalSymlinks(path)
+}
+
+// loopFilePath returns the file path of the mounted filesystem image, backing
+// the given loopback device.
+func loopFilePath(spec string) (string, error) {
+	// if it's not a loopback device, return the input
+	if !strings.Contains(spec, "/dev/loop") {
+		return spec, nil
+	}
+	info, err := getLoopInfo(spec)
+	if err != nil {
+		return "", errwrap.Wrapf(err, "error getting loop info")
+	}
+	// trim the extra null chars off the end of the filename
+	return string(bytes.Trim(info.FileName[:], "\x00")), nil
+}
+
+// loopInfo is a datastructure that holds relevant information about a file
+// backed loopback device. Code is based on freddierice/go-losetup.
+type loopInfo struct {
+	Device         uint64
+	INode          uint64
+	RDevice        uint64
+	Offset         uint64
+	SizeLimit      uint64
+	Number         uint32
+	EncryptType    uint32
+	EncryptKeySize uint32
+	Flags          uint32
+	FileName       [64]byte
+	CryptName      [64]byte
+	EncryptKey     [32]byte
+	Init           [2]uint64
+}
+
+// getLoopInfo returns a loopInfo struct containing information about the
+// provided file backed loopback device.
+func getLoopInfo(loop string) (*loopInfo, error) {
+	// open the loop file
+	f, err := os.OpenFile(loop, 0, loopFileUmask)
+	if err != nil {
+		return nil, fmt.Errorf("error opening %s: %s", loop, err)
+	}
+	defer f.Close()
+
+	// deserialize the contents
+	retInfo := loopInfo{}
```

Comment:
the convention is to just do the pointer upfront. eg: 

`retInfo := &loopInfo{}`

Reply:
It panics? I'm guessing you forgot to remove the & when you pass it in via unsafe() ?
But current commit looks like it's all good?